extensions/investspec/investspec.py

```diff
@@ -0,0 +1,418 @@
+import docutils
+import importlib
+
+INPUT_TYPES_HTML_FILE = 'input_types.html'
+# accepted geometries for a vector will be displayed in this order
+GEOMETRY_ORDER = [
+    'POINT',
+    'MULTIPOINT',
+    'LINESTRING',
+    'MULTILINESTRING',
+    'POLYGON',
+    'MULTIPOLYGON']
+
+
+def format_type_string(arg_type):
+    """Represent an arg type as a user-friendly string.
+
+    Args:
+        arg_type (str): the type to format
+
+    Returns:
+        formatted string that links to a description of the input type
+    """
+    # Represent the type as a string. Some need a more user-friendly name.
+    # we can only use standard docutils features here, so no :ref:
+    # this syntax works to link to a section in a different page, but it
+    # isn't universally supported and depends on knowing the built page name.
+    if arg_type == 'freestyle_string':
+        return f'`text <{INPUT_TYPES_HTML_FILE}#text>`__'
+    elif arg_type == 'option_string':
+        return f'`option <{INPUT_TYPES_HTML_FILE}#option>`__'
+    elif arg_type == 'boolean':
+        return f'`true/false <{INPUT_TYPES_HTML_FILE}#truefalse>`__'
+    else:
+        return f'`{arg_type} <{INPUT_TYPES_HTML_FILE}#{arg_type}>`__'
+
+
+def format_units_string(unit):
+    """Represent a pint Unit as a user-friendly phrase.
+
+    Args:
+        unit (pint.Unit): the unit to format
+
+    Returns:
+        string describing the unit
+    """
+    # pluralize the first unit so that it reads more naturally
+    custom_unit_plurals = {
+        'foot': 'feet',
+        'degree_Celsius': 'degrees_Celsius'
+    }
+    units_string = str(unit)
+    if units_string == 'none':
+        return ''
+
+    words = units_string.split(' ')
+    first_unit = words[0]
+    # check if it has an irregular plural form
+    if first_unit in custom_unit_plurals:
+        words[0] = custom_unit_plurals[first_unit]
+    # for all others, add 's' to the end
+    else:
+        words[0] = words[0] + 's'
+    units_string = ' '.join(words)
+    # pint separates words with underscores
+    units_string = units_string.replace('_', ' ')
+    # represent exponents with a caret rather than asterisks
+    units_string = units_string.replace(' ** ', '^')
+    # remove spaces around slashes
+    units_string = units_string.replace(' / ', '/')
+    return units_string
+
+
+def format_required_string(required):
+    """Represent an arg's required status as a user-friendly string.
+
+    Args:
+        required (bool | str): required property of an arg. May be `True`,
+            `False`, or a conditional string.
+
+    Returns:
+        string
+    """
+    if required is True:
+        return 'required'
+    elif required is False:
+        return 'optional'
+    else:
+        # assume that the about text will describe the conditional
+        return 'conditionally required'
```

Comment:
It's hard to separate what "conditionally" means to us in invest from its everyday meaning. I do feel like it nicely sums up the intent, but "sometimes" is definitely an easier word. Most of the user's guide is written at an advanced level anyway, though...

This brings up some interesting questions about who our audience is and how we can be accessible to them.
Some sections of the UG are only accessible to groups as specific as "grad students with experience in hydrology modeling" (or educated fluent English speakers who can look up all the jargon). What does the priority  _"Expand the user base of InVEST to include users outside of research"_ mean for the user's guide? Should it be understandable to high school graduates? college graduates in any field? and what is the equivalent for non-native speakers?

We can distinguish between words/phrases that are difficult because
(1) they are technical jargon
(2) they are idioms or specific to American English
(3) they are advanced vocabulary (but not specific to a field or region)
IMO, we should try to avoid (2), but (1) and (3) are okay to a certain extent. 

This [readability calculator](https://readabilityformulas.com/free-readability-formula-tests.php) rated a section of the CBC users' guide at an 11th grade reading level. There are some interesting resources for this sort of thing: https://readable.com/ https://datayze.com/word-analyzer

Reply:
This is an important point about who our audience is and to whom we'd like to expand the user base. The general view from the science team is that the UG tends to be too technical, especially the chapters on terrestrial models.  I would personally advocate for at least a high school graduate readability access (assuming familiarity with science), but of course that would take considerably more work to revamp many chapters and may not be possible for every model.

Reply:
If we were to revamp a lot of chapters like you suggest, it would be great to get help from a professional technical writer.

Reply:
Hmmm, true, but my sense is that may not be in the budget? It sounds like it's up to Stacie and myself for now, as we have time.

Reply:
Thanks for this context James!